torch/storage.py

```diff
@@ -363,8 +363,10 @@ def is_hpu(self):
 
     @property
     def filename(self) -> _Optional[str]:
-        """Returns the file name associated with this storage if the storage was memory mapped from a file.
-           or ``None`` if the storage was not created by memory mapping a file."""
+        """Returns the file name associated with this storage.
+
+        The file name will be a string if the storage is on CPU and was either created
+        via :meth:`~torch.from_file()` with ``shared`` as ``True``. This attribute is ``None`` otherwise."""
```

Comment:
You're missing the "or" for the "either"?

Reply:
oops there should be no either, thanks